### ISSUE TITLE
Installation documentation

### ISSUE THREAD

**Issue by jmatsushita** [ImaginationForPeople/assembl#8]
Hi there,

I'm about to attempt to install Assembl again but on a dedicated VPS (instead of using docker #3). I will follow the developer installation notes, but are there other step by step instructions?

Cheers,

Jun

**Comment by maparent** [comment]
The README is what we use ourselves. Do let us know if you run into any snags, we'll definitely want to update it.

**Comment by jmatsushita** [comment]
Here are my installation notes.

On a fresh Ubuntu 14.04.1 LTS here's what worked for me (indicating what was missing in the current instructions):

These 3 steps where missing leading to fabric not being found.
```
apt-get update
apt-get install software-properties-common python-software-properties
add-apt-repository universe
```

```
apt-get install fabric git openssh-server
```

I then created an assembl user.
```
adduser assembl
addgroup assembl sudo
su - assembl
```

Then
```
git clone https://github.com/ImaginationForPeople/assembl.git
cd assembl
fab devenv install_builddeps
fab devenv bootstrap_from_checkout
```

It's installing now I'll report back on the rest of the adventure.

**Comment by maparent** [comment]
OK. We'll check about the universe repository and add that. As for the user, we're running all processes under www-data ourselves. Creating a user makes sense, but might also complicate things later on. Let us know.

**Comment by jmatsushita** [comment]
So it seems that after the boostrap_from_checkout supervisord actually started so when going through the venv activation and running supervisord I got:

```
~/assembl$ source venv/bin/activate
~/assembl$ supervisord
Error: Another program is already listening on a port that one of our HTTP servers is configured to use.  Shut this program down first before starting supervisord.
For help, use /home/assembl/assembl/venv/bin/supervisord -h
```

Running a ```ps``` showed that things seemed to be already running. So I crated the user. and it seems to have worked but there were a lot of warnings:

```
$ assembl-add-user --email jun@iilab.org --name "Jun Matsushita" --username jun --password jun development.ini
/home/assembl/assembl/venv/local/lib/python2.7/site-packages/sqlalchemy/orm/mapper.py:1599: SAWarning: Property Content.widget_idea_links on Mapper|Content|content being replaced with new property Content.widget_idea_links; the old property will be discarded
  prop,
/home/assembl/assembl/venv/local/lib/python2.7/site-packages/sqlalchemy/orm/mapper.py:1599: SAWarning: Property Content.widget_idea_links on Mapper|Post|post being replaced with new property Content.widget_idea_links; the old property will be discarded
  prop,
/home/assembl/assembl/venv/local/lib/python2.7/site-packages/sqlalchemy/orm/mapper.py:1599: SAWarning: Property Content.widget_idea_links on Mapper|AssemblPost|assembl_post being replaced with new property Content.widget_idea_links; the old property will be discarded
  prop,
/home/assembl/assembl/venv/local/lib/python2.7/site-packages/sqlalchemy/orm/mapper.py:1599: SAWarning: Property Content.widget_idea_links on Mapper|SynthesisPost|synthesis_post being replaced with new property Content.widget_idea_links; the old property will be discarded
  prop,
/home/assembl/assembl/venv/local/lib/python2.7/site-packages/sqlalchemy/orm/mapper.py:1599: SAWarning: Property Content.widget_idea_links on Mapper|WidgetPost|post_with_metadata being replaced with new property Content.widget_idea_links; the old property will be discarded
  prop,
/home/assembl/assembl/venv/local/lib/python2.7/site-packages/sqlalchemy/orm/mapper.py:1599: SAWarning: Property Content.widget_idea_links on Mapper|IdeaProposalPost|idea_proposal_post being replaced with new property Content.widget_idea_links; the old property will be discarded
  prop,
/home/assembl/assembl/venv/local/lib/python2.7/site-packages/sqlalchemy/orm/mapper.py:1599: SAWarning: Property Content.widget_idea_links on Mapper|ImportedPost|imported_post being replaced with new property Content.widget_idea_links; the old property will be discarded
  prop,
/home/assembl/assembl/venv/local/lib/python2.7/site-packages/sqlalchemy/orm/mapper.py:1599: SAWarning: Property Content.widget_idea_links on Mapper|Email|email being replaced with new property Content.widget_idea_links; the old property will be discarded
  prop,
initialized twice: {'avatar.gravatar_default': 'mm', 'celery_tasks.notify.imodeleventwatcher': 'assembl.lib.model_watcher.ModelEventWatcherPrinter', 'public_port': '6543', 'facebook.consumer_key': '', 'pyramid.reload_templates': 'true', 'assembl.domain': 'assembl.net', 'twitter.consumer_secret': '', 'celery_tasks.notification_dispatch.imodeleventwatcher': 'assembl.models.notification.ModelEventWatcherNotificationSubscriptionDispatcher', 'facebook.debug_mode': 'false', 'assembl_debug_signal': 'true', 'sqlalchemy.strategy': 'atexit_cleanup', 'facebook.export_permissions': 'public_profile, email, publish_actions, user_posts, user_likes, manage_pages, publish_pages, user_groups, user_managed_groups', 'supported_exports_list': '', 'cache_viewdefs': 'false', 'store': 'sqlalchemy', 'login_providers': 'facebook twitter google', 'beaker.session.lock_dir': '/home/assembl/assembl/data/sessions/lock', 'celery_tasks.notification_dispatch.broker': 'redis://localhost:6379/0', 'sqlalchemy.url': 'virtuoso://assembl:assembl@VOSU', '__file__': '/home/assembl/assembl/development.ini', 'jinja2.directories': 'assembl:templates', 'sqlalchemy.echo': 'False', 'activate_tour': 'true', 'dogpile_cache.backend': 'file', 'available_languages': 'fr_CA en_CA', 'celery_tasks.imap.imodeleventwatcher': 'assembl.tasks.notification_dispatch.ModelEventWatcherCelerySender', 'dogpile_cache.arguments.filename': 'var/dogpile_cache.dbm', 'mail.host': 'localhost', 'facebook.scope': 'public_profile,email', 'debugtoolbar.hosts': '127.0.0.0/8', 'session.cookie_on_exception': 'true', 'web_analytics_piwik_script': '<!-- Piwik -->\n<script type="text/javascript">\nvar _paq = _paq || [];\n_paq.push([\'trackPageView\']);\n_paq.push([\'enableLinkTracking\']);\n(function() {\nvar u="//piwik.coeus.ca/";\n_paq.push([\'setTrackerUrl\', u+\'piwik.php\']);\n_paq.push([\'setSiteId\', %d]);\nvar d=document, g=d.createElement(\'script\'), s=d.getElementsByTagName(\'script\')[0];\ng.type=\'text/javascript\'; g.async=true; g.defer=true; g.src=u+\'piwik.js\'; s.parentNode.insertBefore(g,s);\n})();\n</script>\n<noscript><p><img src="//piwik.coeus.ca/piwik.php?idsite=%d" style="border:0;" alt="" /></p></noscript>\n<!-- End Piwik Code -->', 'google.scope': 'profile', 'assembl_handle_exceptions': 'false', 'assembl.imodeleventwatcher': 'assembl.tasks.notification_dispatch.ModelEventWatcherCelerySender', 'google.consumer_key': '', 'trusted_login_providers': 'facebook google twitter', 'db_password': 'assembl', 'changes.multiplex': 'true', 'require_secure_connection': 'false', 'handle_exceptions': '', 'debugtoolbar.intercept_exc': 'display', 'db_schema': 'assembl', 'public_hostname': 'localhost', 'beaker.session.type': 'ext:memcached', 'github.scope': 'repo', 'session.secret': 'CHANGEME_enter_a_secret', 'requirejs.cache_bust': 'true', 'celery_tasks.notify.broker': 'redis://localhost:6379/0', 'pyramid.debug_authorization': 'false', 'tm.attempts': '2', 'celery_tasks.imap.broker': 'redis://localhost:6379/0', 'subscriptions.participant.default': 'FOLLOW_SYNTHESES\nEMAIL_BOUNCED\nEMAIL_VALIDATE\nPARTICIPATED_FOR_FIRST_TIME_WELCOME\nSUBSCRIPTION_WELCOME', 'dogpile_cache.expiration_time': '10000', 'assembl.validate_registration_emails': 'false', 'accept_secure_connection': 'false', 'pyramid.includes': '\npyramid_mailer.debug\npyramid_debugtoolbar\npyramid_debugtoolbar_ajax\npyramid_tm', 'here': '/home/assembl/assembl', 'twitter.consumer_key': '', 'source_reader_debug_signal': 'true', 'beaker.session.url': '127.0.0.1:11211', 'security.email_token_salt': '[enter a key]', 'google.consumer_secret': '', 'assembl.admin_email': 'webmaster@assembl.net', 'help_url': 'http://assembl.org/user-guides/', 'pyramid.debug_routematch': 'false', 'sqltap': 'true', 'facebook.app_access_token': '', 'changes.socket': 'ipc:///tmp/assembl_changes/0', 'store.url': 'sqlite:////home/assembl/assembl/assembl.db', 'session.key': 'CHANGEME_enter_a_key', 'changes.prefix': '/socket', 'minified_js': 'false', 'facebook.api_version': '', 'db_user': 'assembl', 'pyramid.debug_notfound': 'false', 'changes.websocket.port': '8085', 'facebook.consumer_secret': '', 'pyramid.default_locale_name': 'en_CA', 'changes.websocket.url': 'http://localhost:8085/socket/', 'beaker.session.data_dir': '/home/assembl/assembl/data/sessions/data', 'debugtoolbar.enabled': 'true'}
keeping: {'assembl.imodeleventwatcher': 'assembl.tasks.notification_dispatch.ModelEventWatcherCelerySender', 'google.consumer_key': '', 'pyramid.debug_routematch': 'false', 'available_languages': 'fr_CA en_CA', 'beaker.session.url': '127.0.0.1:11211', 'db_password': 'assembl', 'celery_tasks.notification_dispatch.broker': 'redis://localhost:6379/0', 'pyramid.reload_templates': 'true', 'changes.multiplex': 'true', 'accept_secure_connection': 'false', 'require_secure_connection': 'false', 'handle_exceptions': '', 'debugtoolbar.intercept_exc': 'display', 'db_schema': 'assembl', 'public_hostname': 'localhost', 'facebook.consumer_secret': '', 'beaker.session.type': 'ext:memcached', 'github.scope': 'repo', 'facebook.consumer_key': '', 'assembl.domain': 'assembl.net', 'config_uri': '/home/assembl/assembl/development.ini', 'session.secret': 'CHANGEME_enter_a_secret', 'requirejs.cache_bust': 'true', 'twitter.consumer_secret': '', 'celery_tasks.notify.imodeleventwatcher': 'assembl.lib.model_watcher.ModelEventWatcherPrinter', 'dogpile_cache.arguments.filename': 'var/dogpile_cache.dbm', 'pyramid.debug_authorization': 'false', 'public_port': '6543', 'celery_tasks.imap.broker': 'redis://localhost:6379/0', 'subscriptions.participant.default': 'FOLLOW_SYNTHESES\nEMAIL_BOUNCED\nEMAIL_VALIDATE\nPARTICIPATED_FOR_FIRST_TIME_WELCOME\nSUBSCRIPTION_WELCOME', 'facebook.debug_mode': 'false', 'assembl_handle_exceptions': 'false', 'assembl_debug_signal': 'true', 'dogpile_cache.expiration_time': '10000', 'sqlalchemy.strategy': 'atexit_cleanup', 'facebook.export_permissions': 'public_profile, email, publish_actions, user_posts, user_likes, manage_pages, publish_pages, user_groups, user_managed_groups', 'supported_exports_list': '', 'cache_viewdefs': 'false', 'store': 'sqlalchemy', 'google.scope': 'profile', 'beaker.session.lock_dir': '/home/assembl/assembl/data/sessions/lock', 'assembl.validate_registration_emails': 'false', 'trusted_login_providers': 'facebook google twitter', 'pyramid.includes': '\npyramid_mailer.debug\npyramid_debugtoolbar\npyramid_debugtoolbar_ajax\npyramid_tm', 'sqlalchemy.url': 'virtuoso://assembl:assembl@VOSU', 'assembl.admin_email': 'webmaster@assembl.net', 'jinja2.directories': 'assembl:templates', 'sqlalchemy.echo': 'False', 'google.consumer_secret': '', 'twitter.consumer_key': '', 'source_reader_debug_signal': 'true', 'activate_tour': 'true', 'dogpile_cache.backend': 'file', 'tm.attempts': '2', 'security.email_token_salt': '[enter a key]', 'avatar.gravatar_default': 'mm', 'help_url': 'http://assembl.org/user-guides/', 'celery_tasks.imap.imodeleventwatcher': 'assembl.tasks.notification_dispatch.ModelEventWatcherCelerySender', 'celery_tasks.notification_dispatch.imodeleventwatcher': 'assembl.models.notification.ModelEventWatcherNotificationSubscriptionDispatcher', 'sqltap': 'true', 'mail.host': 'localhost', 'facebook.app_access_token': '', 'facebook.scope': 'public_profile,email', 'changes.socket': 'ipc:///tmp/assembl_changes/0', 'debugtoolbar.hosts': '127.0.0.0/8', 'session.key': 'CHANGEME_enter_a_key', 'changes.prefix': '/socket', 'login_providers': 'facebook twitter google', 'session.cookie_on_exception': 'true', 'minified_js': 'false', 'facebook.api_version': '', 'web_analytics_piwik_script': '<!-- Piwik -->\n<script type="text/javascript">\nvar _paq = _paq || [];\n_paq.push([\'trackPageView\']);\n_paq.push([\'enableLinkTracking\']);\n(function() {\nvar u="//piwik.coeus.ca/";\n_paq.push([\'setTrackerUrl\', u+\'piwik.php\']);\n_paq.push([\'setSiteId\', %d]);\nvar d=document, g=d.createElement(\'script\'), s=d.getElementsByTagName(\'script\')[0];\ng.type=\'text/javascript\'; g.async=true; g.defer=true; g.src=u+\'piwik.js\'; s.parentNode.insertBefore(g,s);\n})();\n</script>\n<noscript><p><img src="//piwik.coeus.ca/piwik.php?idsite=%d" style="border:0;" alt="" /></p></noscript>\n<!-- End Piwik Code -->', 'db_user': 'assembl', 'pyramid.debug_notfound': 'false', 'changes.websocket.port': '8085', 'celery_tasks.notify.broker': 'redis://localhost:6379/0', 'pyramid.default_locale_name': 'en_CA', 'store.url': 'sqlite:////home/assembl/assembl/assembl.db', 'changes.websocket.url': 'http://localhost:8085/socket/', 'beaker.session.data_dir': '/home/assembl/assembl/data/sessions/data', 'debugtoolbar.enabled': 'true'}
ERROR: Initialized twice.
/home/assembl/assembl/venv/local/lib/python2.7/site-packages/sqlalchemy/orm/relationships.py:1729: SAWarning: Warning: relationship 'discussion' on mapper 'Mapper|Extract|extract' supersedes the same relationship on inherited mapper 'Mapper|IdeaContentPositiveLink|idea_content_positive_link'; this can cause dependency issues during flush
  % (self.key, self.parent, inheriting))
/home/assembl/assembl/venv/local/lib/python2.7/site-packages/sqlalchemy/orm/relationships.py:1729: SAWarning: Warning: relationship 'discussion' on mapper 'Mapper|Extract|extract' supersedes the same relationship on inherited mapper 'Mapper|IdeaContentLink|idea_content_link'; this can cause dependency issues during flush
  % (self.key, self.parent, inheriting))
/home/assembl/assembl/venv/local/lib/python2.7/site-packages/sqlalchemy/orm/relationships.py:1729: SAWarning: Warning: relationship 'discussion' on mapper 'Mapper|TableOfContents|table_of_contents' supersedes the same relationship on inherited mapper 'Mapper|IdeaGraphView|idea_graph_view'; this can cause dependency issues during flush
  % (self.key, self.parent, inheriting))
sent 0 * [{'@id': 'local:UserRole/1', '@private': 'local:AgentProfile/1', 'role': 'r:sysadmin', '@type': 'UserRole'}, {'@id': 'local:AgentProfile/1', 'verified': True, '@type': 'User', 'creation_date': '2015-08-26T15:55:18.092282'}, {'username': 'jun', '@private': 'local:AgentProfile/1', 'verified': True, 'name': 'Jun Matsushita', 'avatar_url_base': '/user/id/1/avatar/', 'creation_date': '2015-08-26T15:55:18.092282', 'accounts': [], 'real_name': 'Jun Matsushita', '@id': 'local:AgentProfile/1', '@type': 'User', '@view': 'private', 'permissions': {}}]
```

Then I sort of ran out of instructions from the README.md

Looking into the development.ini file it seems that the server should be running on ```public_port = 6543``` but there doesn't seem to be any service running there. On the other hand, virtuoso is up and accessible on 8892.

So I saw in the requirements that nginx was there, and it turns out that nginx wasn't installed, so I installed it.

Trying to run the followingfrom the venv didn't help. 
```
supervisorctl start dev:
```

Until I realised that nginx wasn't a dependency for the dev environment and that the ```start dev:``` command actually had brought up the site on port 6543.

How could I switch to a prod setting now with nginx in the front?

Will also open a new issue around configuration now,

**Comment by maparent** [comment]
Those warnings are normal. As for nginx, yes we assume that installing/documenting nginx is out of our documentation's scope; but we need still to say that explicitly. There are sample configuration files in doc/sample_nginx_config and doc/sample_upstart_script .

**Comment by jmatsushita** [comment]
Ok I'll take a look. I guess it might make sense to not have the details of the nginx installation (as you say well documented elsewhere), but it would probably also make sense to have an "Installation in production" section (in addition to the existing "Installation for developers" section) where you could list the high level steps for prod deployment without having to tease it out of the development install instructions (for instance with the sample config you mentioned, or the local.ini trick in issue #10).
